pkg/operator/controller/ingress/status.go

```diff
@@ -874,12 +874,29 @@ func computeDNSStatus(ic *operatorv1.IngressController, wildcardRecord *iov1.DNS
 	case len(wildcardRecord.Status.Zones) > 0:
 		var failedZones []configv1.DNSZone
 		for _, zone := range wildcardRecord.Status.Zones {
-			for _, cond := range zone.Conditions {
-				if cond.Type == iov1.DNSRecordFailedConditionType && cond.Status == string(operatorv1.ConditionTrue) {
-					// check to see if the zone is in the dnsConfig.Spec
-					// fix:BZ1942657 - relates to status changes when updating DNS PrivateZone config
-					if checkZoneInConfig(dnsConfig, zone.DNSZone) {
-						failedZones = append(failedZones, zone.DNSZone)
+			if len(zone.Conditions) > 0 {
+
+				// When scope of load balancer is changed from internal to external
```

Comment:
+1 I really like comments like these that explain why; saves me so much time reviewing.